R/zchunk_LB1322.Fert.R

```diff
@@ -41,92 +43,396 @@ module_energy_LB1322.Fert_DISABLED <- function(command, ...) {
+      mutate(in_Fert_adj = Fert_Prod_MtN_adj * intensity_GJkgN) %>% # Adjust in_Fert with adjusted production
+      # Recheck total energy inputs to fertilizer against industrial energy use for each region
+           # Aggregate to GCAM region
+      group_by(GCAM_region_ID, fuel, year) %>%
+      summarise(Fert_Prod_MtN_adj = sum(Fert_Prod_MtN_adj),
+                in_Fert_adj = sum(in_Fert_adj)) %>%
+      ungroup() %>%
+      # Expand table to write out to all region x fuel x year combinations
+      complete(GCAM_region_ID = list_GCAM_region_ID,
+               fuel = list_fuels,
+               year = HISTORICAL_YEARS,
+               fill = list(in_Fert_adj = 0, Fert_Prod_MtN_adj = 0)) %>%
+      # Re-checking total energy inputs to fertilizer against industrial energy use for each region
+      left_join_error_no_match(L1322.Fert_ALL_MtN_R_F_Y, by = c("GCAM_region_ID", "fuel", "year")) %>%
+      mutate(check = in_indenergy + in_indfeed - in_Fert_adj) %>% # Recheck, adjusting for in_Fert_adj
+      mutate(check = replace(check, abs(check) < 1e-6, 0)) %>%
+      # Recalculate intensity
+      mutate(intensity_GJkgN = in_Fert_adj / Fert_Prod_MtN_adj) %>%
+      replace_na(list(intensity_GJkgN = 0)) %>%
+      # Adding column for feedstock energy density, which was calcuated earlier using hydrogen density
+      mutate(feedstock_GJkgN = NH3_energy_GJkgN) %>% # Adding single column with energy density of feedstock
+      mutate(energy_GJkgN = intensity_GJkgN - feedstock_GJkgN) %>% # Substract energy density of feedstock
+      # Calculate first-order estimate of energy and feedstock quantities in the industrial sector
+      mutate(indfeed_to_Fert = Fert_Prod_MtN_adj * feedstock_GJkgN,
+             indenergy_to_Fert = Fert_Prod_MtN_adj * energy_GJkgN) %>%
+      # Calculate remaining industrial energy and feedstock consumption re-allocating to avoid negative values
+      mutate(in_indenergy_netFert = in_indenergy - indenergy_to_Fert,
+             in_indfeed_netFert = in_indfeed - indfeed_to_Fert) %>%
+      # This is complicated. This last step pertains to the available energy and feedstock quantities for the industrial sector, in regions
+      # where either went negative as a result of including the fertilizer industry (using the IEA's shares of production by fuel type,
+      # modified so that no regions went negative, and following the IEA's conventions on disaggregation of feedstocks and energy-use.
+      # We are assuming that for the purposes of re-allocation of energy from the (general) industrial sector to the fertilizer sector, all energy
+      # is available, whether initially classified as feedstocks or energy-use. This may zero out both energy and feedstocks or both in the
+      # general industrial sector, causing all industrial demands of natural gas to be for the ammonia industry.
+      # Note that the quantities are added because one is negative.
```

Comment:
I believes this makes sense.  I would remove the first sentence about it being and add something that says that the purpose here is to handle the situation where all industrial energy and feedstocks for a country are assigned to ammonia production.